src/sentry/sentry_metrics/querying/metadata/metric_spans.py

```diff
@@ -189,11 +207,23 @@ def _get_spans(
         min_value: Optional[float],
         max_value: Optional[float],
     ) -> Sequence[Span]:
-        # TODO: implement transaction.duration handling by looking for:
-        #   is_segment: 1
-        #   duration >= min and duration <= max
-        #   sentry_tags and tags as values from the query string
-        return []
+        transformed_conditions = transform_to_tags(conditions=conditions, check_sentry_tags=True)
+        transformed_conditions = transform_conditions_with(
```

Comment:
We have to run these mappings in order to generate the correct conditions.

Reply:
It can definitely be done more efficiently in a single pass and without mutating the tree but to keep it modular and easy to improve, I opted to make it separate.